### ISSUE TITLE
how to create `SASLProvider` for multiple token based sasl ?

### ISSUE THREAD

**Issue by Kavan72** [horazont/aioxmpp#361]

**Comment by horazont** [comment]
That is not how normal [SASL](https://tools.ietf.org/html/rfc4422) works. Is there any internet standard where this is specified?

**Comment by Kavan72** [comment]
nope, i've found this on pc game call warface. is this possible too create custom SASLProvider for this scenario ?

**Comment by horazont** [comment]
No. This is not SASL conformant, hence, you cannot write a SASL implementation.

You could still subclass aioxmpp.security_layer.SASLProvider and mimic its interface and see how far you get with that. However, this is no configuration which will be supported in any way.
